wafer/registration/sso.py

```diff
@@ -0,0 +1,146 @@
+        'client_secret': settings.WAFER_GITHUB_CLIENT_SECRET,
+        'code': code,
+    })
+    if r.status_code != 200:
+        raise SSOError('Invalid code')
+    token = r.content
+
+    r = requests.get('https://api.github.com/user?%s' % token)
+    if r.status_code != 200:
+        raise SSOError('Failed response from GitHub')
+    gh = r.json()
+
+    try:
+        login = gh['login']
+        name = gh['name'].partition(' ')[::2]
+    except KeyError:
+        log.debug('Error creating account from github information: %s', e)
+        raise SSOError('GitHub profile missing required content')
+
+    email = gh.get('email', None)
+    if not email:  # No public e-mail address
+        r = requests.get('https://api.github.com/user/emails?%s' % token)
+        if r.status_code != 200:
+            raise SSOError('Failed response from GitHub')
+        try:
+            email = r.json()[0]['email']
+        except (KeyError, IndexError) as e:
+            log.debug('Error extracting github email address: %s', e)
+            raise SSOError('Failed to obtain email address from GitHub')
+
+    profile_fields = {
+        'github_username': login,
+    }
+    if 'blog' in gh:
+        profile_fields['blog'] = gh['blog']
+
+    user = sso(
+        identifier={'userprofile__github_username': login},
+        desired_username=login, name=name, email=email,
+        profile_fields=profile_fields)
+
+    if not user.is_active:
+        raise SSOError('Account disabled')
+    return user
+
+
+def debian_sso(meta):
+    authentication_status = meta.get('SSL_CLIENT_VERIFY', None)
+    if authentication_status != "SUCCESS":
+        raise SSOError('Requires authentication via Client Certificate')
+
+    email = meta['SSL_CLIENT_S_DN_CN']
+    identifier = {'email': email}
+    username = email.split('@', 1)[0]
+
+    name = ('Unknown User', email)
+    if not get_user_model().objects.filter(**identifier).exists():
+        r = requests.get('https://nm.debian.org/api/people',
+                         params={'uid': username},
+                         headers={'Api-Key': settings.WAFER_DEBIAN_NM_API_KEY})
+        if r.status_code != 200:
+            raise SSOError('Failed to query nm.debian.org')
+        if 'r' not in r.json():
+            raise SSOError('Failed to parse nm.debian.org respnose')
+        # The API performs substring queries, so we need to find the correct
+        # entry in the response.
+        for person in r.json()['r']:
+            if person['uid'] == username:
+                first_name = person['cn']
+                if person['mn']:
+                    first_name += u' ' + person['mn']
+                last_name = person['sn']
+                name = (first_name, last_name)
+                break
+
+    user = sso(identifier=identifier, desired_username=username, name=name,
+               email=email)
+
+    if not user.is_active:
+        raise SSOError('Account disabled')
```

Comment:
Since this "if not user.is_active:' construct is repeated in both implementation, should it not be folded into the sso function? Are there circumstances where sso will be called where this is not an error condition?